.github/workflows/build-images-main-sha.yaml

```diff
@@ -0,0 +1,14 @@
+name: Build SHA image for main branch
+
+on:
+  push:
+    branches: ['main']
+
+jobs:
+  workflow-build:
+    name: Calls build-images-base workflow
+    uses: ./.github/workflows/build-images-base.yaml
+    secrets: inherit
+    with:
+      authorinoVersion: ${{ github.sha }}
+      quayImageExpiry: 2w
```

Comment:
I'm making an assumption that SHA build are equivalent to nightly builds that we want to expire 🤔 

If anyone has a better expiry value, I can update this

Reply:
Until now, SHA builds have not been equivalent to nightly builds. As of today, builds are tagged _additionally_ with the commit SHA on top of the other, more human-friendly tag. This includes `latest` (for `main` branch), feature branches, and long-lived release builds.

I'm not sure how I feel about breaking this link between tags, effectively building two separate images now, if this is only for the purpose of cleaning up the registry.

The `quay.expires-after` label seems easy enough, but it only works for expiring an entire image, not an individual image tag.

Reply:
Hmm, I guess the question is then, does authorino-operator images need / want to expire images through this label? To me, it seems like it may not since the current workflow triggers on merge to main, which would build with an equivalent SHA tag. These SHA tags are kept indefinitely, but maybe that is something we want.

Feature branch and release builds looks to require manual trigger of the same workflow which do not happen that regularly